backends/vllm/src/models/llama.rs

```diff
@@ -78,62 +121,33 @@ impl ModelLoader for LlamaModel {
         info!("Loading Llama model ...");
         let start = Instant::now();
 
-        let (model, config) = {
-            let config: LlamaConfig =
-                serde_json::from_slice(&std::fs::read(&file_paths.config_path)?)?;
-            let config = config.into_config();
-
+        let model = {
             let vb = unsafe {
                 VarBuilder::from_mmaped_safetensors(
                     file_paths.weights_path.as_slice(),
                     dtype,
-                    &device,
+                    device,
                 )?
             };
-            (Llama::load(vb, &config, dtype, &device)?, config)
+            Llama::load(vb, &config, dtype, device)?
         };
         info!("Loaded Llama model in {:?}", start.elapsed());
 
         Ok(Self { model, config })
     }
-}
-
-impl ModelMetadata for LlamaModel {
-    fn alibi_slopes(&self) -> Option<&Tensor> {
-        None
-    }
 
-    fn eos_token_ids(&self) -> Option<Vec<u32>> {
-        match self.config.eos_token_id.clone() {
-            Some(LlamaEosToks::Single(id)) => Some(vec![id]),
-            Some(LlamaEosToks::Multiple(ids)) => Some(ids),
-            None => None,
-        }
-    }
-
-    fn hidden_dim(&self) -> usize {
-        self.config.hidden_size / self.config.num_attention_heads
-    }
-
-    fn num_attention_heads(&self) -> usize {
-        self.config.num_attention_heads
-    }
-
-    fn num_hidden_layers(&self) -> usize {
-        self.config.num_hidden_layers
-    }
-
-    fn num_kv_heads(&self) -> usize {
-        self.config.num_key_value_heads
-    }
-
-    fn softmax_scale(&self) -> f32 {
-        let head_dim = self.hidden_dim();
-        1f32 / (head_dim as f32).sqrt()
-    }
-
-    fn sliding_window(&self) -> Option<usize> {
-        None
+    #[cfg(feature = "nccl")]
```

Comment:
you can probably delete this feature check as well and do one in the mod.rs instead. Because we know you will either use llama.rs or llama_nccl.rs, never both at the same time